### DISCUSSION TITLE
SPA Mode and Cloudflare - Rocket Loader breaks SPA Features

### DISCUSSION THREAD

**Discussion by bturski** [filamentphp/filament#14573] [General]
TLDR; If you're using Cloudflare, be sure to disable Rocket Loader if using SPA mode.

Filament 3.2.119
Laravel 11.28.1
Livewire 3.5.12
PHP 8.3.7

I'm new to Filament (and Livewire) and am working on a Laravel site deployed to Cloud Run. I use other Livewire components and Laravel Pulse. When I switched my panel to SPA mode (https://filamentphp.com/docs/3.x/panels/configuration#spa-mode), I could navigate between resources, but other Livewire features were non-functional (i.e., column sorting, filters). If I refreshed the page, Livewire features would work again until I navigated to a different resource. All Livewire features worked fine running locally, only failed when deployed to Cloud Run.

Because this worked locally but not after deploying, I immediately considered Cloudflare and spent time looking through configuration and enabling developer mode to disable caching. This didn't help so I moved on.

After navigating to a new resource, an error in the console indicated an issue with the Livewire scripts:
![image](https://github.com/user-attachments/assets/0decd1bf-8b39-4a7f-8c01-08ea2cf4f799)

The DOM update showed that the script tags were commented out on the page:
![image](https://github.com/user-attachments/assets/6610c2c6-5069-498d-a88f-28bbe2c34118)

It took much longer than I'd like to admit to notice this:
![image](https://github.com/user-attachments/assets/f37f15f7-87dc-4270-a5d1-e2c3a4fc17f3)

To disable Rocker Loader, you can find it under Speed -> Optimization -> Content Optimization tab -> Rocket Loader (https://developers.cloudflare.com/speed/optimization/content/rocket-loader/enable/)

**Comment by NgYueHong** [comment]
Thank you for sharing.
